pkg/config/database.go

```diff
@@ -31,9 +38,33 @@ func (d *Database) Open(logger *zap.SugaredLogger) (*icingadb.DB, error) {
 		driver.Register(logger)
 	})
 
-	dsn := fmt.Sprintf(
-		"%s:%s@tcp(%s:%d)/%s?timeout=60s",
-		d.User, d.Password, d.Host, d.Port, d.Database)
+	myCfg := mysql.NewConfig()
+
+	myCfg.User = d.User
+	myCfg.Passwd = d.Password
+	myCfg.Net = "tcp"
+	myCfg.Addr = net.JoinHostPort(d.Host, fmt.Sprint(d.Port))
+	myCfg.DBName = d.Database
+	myCfg.Timeout = time.Minute
+
+	tlsConfig, err := d.TLS.MakeConfig(myCfg.Addr)
+	if err != nil {
+		return nil, err
+	}
+
+	if tlsConfig != nil {
+		hash := sha1.New()
+		if err := objectpacker.PackAny([]string{d.TLS.Cert, d.TLS.Key, d.TLS.Ca}, hash); err != nil {
+			return nil, errors.Wrap(err, "can't pack TLS config")
+		}
```

Comment:
> Having is better than needing.

– Eric

Reply:
Well, if we'd actually need this at some point, we'd probably need something more reliable than this anyways. What's hashed here doesn't match what's actually set in `tls.Config`. The hash misses the `ServerName` and `InsecureSkipVerify` flags, so if at some point in the future, we'd have the need for multiple MySQL connections, this could potentially lead to certificates being verified against the wrong server name or not verified at all.

And please don't fix is by adding these two values to the hash now. That only has the potential that support for yet another TLS option is added to the struct in the future but is forgotten here, introducing the same issue again.

I think a static string would totally suffice for now, we don't need an insufficient solution for an issue we don't have. But if you insist on implementing something to support different configurations, at least use something reliable, like for example an unique counter to identify individual configurations.